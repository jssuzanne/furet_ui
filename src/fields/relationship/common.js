/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import {dispatchAll} from '../../store';
import {json_post} from '../../server-call';
import _ from 'underscore';

export const RelationShip = {
    methods: {
        format (condition, fields) {
            return eval(condition);
        },
        addInBreadscrumb (options) {
            const changes = Object.assign({}, this.$store.state.data.changes);
            const route = this.$route;
            const action = this.$store.state.data.actions[String(route.params.actionId)];
            this.$store.commit('ADD_IN_BREADSCRUMB', {
                path: route.path,
                label: action.label,
                changes,
            });
            const params = {
                spaceId: options.spaceId || route.params.spaceId,
                menuId: options.menuId,
                dataId: options.dataId,
                mode: options.mode || 'readonly',
            }
            json_post('/action/' + options.actionId, params, {
                onSuccess: (results) => {
                    dispatchAll(results);
                },
            })
            this.$store.commit('CLEAR_ALL_CHANGE', {});
        },
    },
}

export const RelationShipX2MList = {
    template: `
        <span v-if="isInvisible" />
        <div v-else>
            <span 
                v-for="value in values"
                class="tag" 
                v-bind:style="getStyle(value.dataId)"
            >
                <a 
                    v-on:click.stop="onClick(value.dataId)">{{value.label}}
                </a>
            </span>
        </div>`,
    computed: {
        values () {
            const values = this.row[this.header.name] || '';
            const model = this.header.model; 
            if (model) {
                let data = this.$store.state.data.data;
                if (data[model]) {
                    data = data[model];
                    return _.map(values, dataId => {
                        return {dataId, label: this.format(this.header.display, data[String(dataId)])}
                    })
                }
            }
            return [];
        },
    },
    methods: {
        getStyle (dataId) {
            if (this.header.fieldcolor) {
                const data = this.$store.state.data.data[this.header.model][dataId];
                if (data[this.header.fieldcolor]) return {border: '4px solid ' + data[this.header.fieldcolor]};
            }
            return {};
        },
        onClick (dataId) {
            this.addInBreadscrumb({
                spaceId: this.header.spaceId,
                menuId: this.header.menuId,
                actionId: this.header.actionId,
                dataId,
                mode: this.header.mode,
            });
        },
    },
}

export const RelationShipX2MThumbnail = {
    props: ['model', 'spaceId', 'menuId', 'actionId', 'mode', 'display', 'fieldcolor'],
    template: `
        <div v-if="this.isInvisible" />
        <b-tooltip 
            v-bind:label="getTooltip" 
            v-bind:position="tooltipPosition"
            v-else
        >
            <b-field 
                v-bind:label="this.label"
                v-bind:style="{'width': 'inherit'}"
            >
                <span 
                    v-for="value in values"
                    class="tag" 
                    v-bind:style="getStyle(value.dataId)"
                >
                    <a 
                        v-on:click.stop="onClick(value.dataId)">{{value.label}}
                    </a>
                </span>
            </b-field>
        </b-tooltip>`,
    computed: {
        values () {
            const values = this.data && this.data[this.name] || '';
            if (this.model) {
                let data = this.$store.state.data.data;
                if (data[this.model]) {
                    data = data[this.model];
                    return _.map(values, dataId => {
                        return {dataId, label: this.format(this.display, data[String(dataId)])}
                    })
                }
            }
            return [];
        }
    },
    methods: {
        getStyle (dataId) {
            if (this.fieldcolor) {
                const data = this.$store.state.data.data[this.model][dataId];
                if (data[this.fieldcolor]) return {border: '4px solid ' + data[this.fieldcolor]};
            }
            return {};
        },
        onClick (dataId) {
            this.addInBreadscrumb({
                spaceId: this.spaceId,
                menuId: this.menuId,
                actionId: this.actionId,
                dataId: dataId,
                mode: this.mode,
            });
        },
    },
}

export const RelationShipX2MForm = {
    props: ['model', 'spaceId', 'menuId', 'actionId', 'mode', 'display', 'fieldcolor'],
    computed: {
        values () {
            const values = this.config && this.config.data && this.config.data[this.name] || '';
            if (this.model) {
                let data = this.$store.state.data.data;
                if (data[this.model]) {
                    data = data[this.model];
                    return _.map(values, dataId => {
                        return {dataId, label: this.format(this.display, data[String(dataId)])}
                    })
                }
            }
            return [];
        }
    },
    methods: {
        getStyle (dataId) {
            if (this.fieldcolor) {
                const data = this.$store.state.data.data[this.model][dataId];
                if (data[this.fieldcolor]) return {border: '4px solid ' + data[this.fieldcolor]};
            }
            return {};
        },
        onClick (dataId) {
            this.addInBreadscrumb({
                spaceId: this.spaceId,
                menuId: this.menuId,
                actionId: this.actionId,
                dataId: dataId,
                mode: this.mode,
            });
        },
    },
}
