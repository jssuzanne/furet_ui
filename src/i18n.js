/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
export default {
    locale: 'en',
    fallbackLocale: 'en',
    messages: {
        en: {
            menus: {
                close: 'Close',
                search: 'Search ...',
            },
            views: {
                unknown: {
                    title: 'The wanted view "{name}" is unknown',
                    message: "Report this message to the administrator",
                },
                common: {
                    create: 'Create',
                    save: 'Save',
                    edit: 'Edit',
                    cancel: 'Cancel',
                    delete: 'Delete',
                    close: 'Close',
                    actions: 'Actions',
                    more: 'More',
                },
                clients: {
                    login: {
                        button: 'Log in',
                    },
                },
            },
        },
    },
}
