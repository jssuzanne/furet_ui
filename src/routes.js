/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import VueRouter from 'vue-router';
Vue.use(VueRouter);
import {dispatchAll} from './store';
import {json_post} from './server-call';


const spaceChildren = [
    {
        path: 'action/:actionId',
        props: true,
        component: {
            template: '<furet-ui-space v-bind:actionId="actionId" />',
            props: ['actionId'],
        },
        beforeEnter: (to, from, next) => {
            next()
        },
        children: [
            {
                path: 'view/:viewId',
                props: true,
                component: {
                    template: '<furet-ui-space v-bind:viewId="viewId" />',
                    props: ['viewId'],
                },
                beforeEnter: (to, from, next) => {
                    console.log('view', to, from)
                    next()
                },
            },
        ],
    },
];



const router = new VueRouter({
    routes: [
        {
            path: '/custom/view/:viewName',
            name: 'custom_view',
            props: true,
            component: {
                template: '<furet-ui-custom-view v-bind:viewName="viewName" />',
                props: ['viewName'],
                beforeRouteEnter: (to, from, next) => {
                    json_post('/custom/view/' + to.params.viewName, {}, {
                        onSucess: (results) => {
                            dispatchAll(undefined, results);
                        },
                    })
                    next();
                },
                beforeRouteUpdate: (to, from, next) => {
                    json_post('/custom/view/' + to.params.viewName, {}, {
                        onSucess: (results) => {
                            dispatchAll(undefined, results);
                        },
                    })
                    next();
                },
            },
        },
        {
            path: '/space/:spaceId',
            name: 'space',
            props: true,
            component: {
                template: '<furet-ui-space v-bind:spaceId="spaceId" />',
                props: ['spaceId'],
                beforeRouteEnter: (to, from, next) => {
                    call('/space/' + to.params.spaceId, to.params);
                    next();
                },
                beforeRouteUpdate: (to, from, next) => {
                    if (to.params.spaceId != from.params.spaceId) call('/space/' + to.params.spaceId, to.params);
                    next();
                },
            },
            children: spaceChildren,
        },
        {
            path: '/space/:spaceId/menu/:menuId',
            name: 'space_menu',
            props: true,
            component: {
                template: '<furet-ui-space v-bind:spaceId="spaceId" v-bind:menuId="menuId" />',
                props: ['spaceId', 'menuId'],
                beforeRouteEnter: (to, from, next) => {
                    console.log('plop', to.params)
                    call('/space/' + to.params.spaceId, to.params);
                    next();
                },
                beforeRouteUpdate: (to, from, next) => {
                    console.log('plop', to.params)
                    if (to.params.spaceId != from.params.spaceId) call('/space/' + to.params.spaceId, to.params);
                    next();
                },
            },
            children: [
                {
                    path: 'action/:actionId',
                    name: 'space_menu_action',
                    props: true,
                    component: {
                        template: '<div>{{spaceId}} {{menuId}} {{actionId}} </div>',
                        props: ['spaceId', 'menuId', 'actionId'],
                    },
                    beforeEnter: (to, from, next) => {
                        console.log('action', to, from)
                        next()
                    },
                    children: [
                        {
                            path: 'view/:viewId',
                            props: true,
                            component: {
                                template: '<furet-ui-space v-bind:viewId="viewId" />',
                                props: ['viewId'],
                            },
                            beforeEnter: (to, from, next) => {
                                console.log('view', to, from)
                                next()
                            },
                        },
                    ],
                },
            ],
            // children: [
            //     {
            //         path: 'action/:actionId',
            //         name: 'space_menu_action',
            //         props: true,
            //         component: {
            //             template: '<furet-ui-space v-bind:spaceId="spaceId" v-bind:menuId="menuId" v-bind:actionId="actionId"/>',
            //             props: ['spaceId', 'menuId', 'actionId'],
            //             beforeRouteEnter: (to, from, next) => {
            //                 console.log('action.beforeRouteEnter', to)
            //                 next();
            //             },
            //             beforeRouteUpdate: (to, from, next) => {
            //                 console.log('action.beforeRouteUpdate', to)
            //                 next();
            //             },
            //         }
            //     }
            // ],
        },
    ],
});

const call = (path, params) => {
    json_post(path, params, {
        onSuccess: (results) => {
            dispatchAll(router, results);
        },
    })
}

export default router;
