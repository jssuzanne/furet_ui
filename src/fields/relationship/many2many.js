/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import {FormMixin, ThumbnailMixin, ListMixin} from '../common';
import {RelationShip, RelationShipX2MList, RelationShipX2MThumbnail} from './common';
import {dispatchAll} from '../../store';
import {json_post} from '../../server-call';

export const FieldListMany2Many = Vue.component('furet-ui-list-field-many2many', {
    mixins: [ListMixin, RelationShip, RelationShipX2MList],
})

export const FieldThumbnailMany2Many = Vue.component('furet-ui-thumbnail-field-many2many', {
    mixins: [ThumbnailMixin, RelationShip, RelationShipX2MThumbnail],
})
