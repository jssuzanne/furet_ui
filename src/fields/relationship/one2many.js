/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import {FormMixin, ThumbnailMixin, ListMixin} from '../common';
import {RelationShip, RelationShipX2MList, RelationShipX2MThumbnail} from './common';
import {dispatchAll} from '../../store';
import {json_post} from '../../server-call';

export const FieldListOne2Many = Vue.component('furet-ui-list-field-one2many', {
    mixins: [ListMixin, RelationShip, RelationShipX2MList],
})

export const FieldThumbnailOne2Many = Vue.component('furet-ui-thumbnail-field-one2many', {
    mixins: [ThumbnailMixin, RelationShip, RelationShipX2MThumbnail],
})

export const View = Vue.component('furet-ui-x2m-view', {
    props: ['model', 'views', 'viewId', 'dataIds', 'dataId', 'isReadonly', 'x2oField', 'x2oFieldId'],
    render: function(createElement) {
        let view = plugin.get(['views', 'x2m-type', this.view.viewType]);
        if (!view) view = plugin.get(['views', 'Unknown']).vue;
        return createElement(view, {
            props: {
                model: this.model,
                views: this.views,
                viewId: this.viewId,
                view: this.view,
                viewName: this.view.viewType,
                dataIds: this.dataIds,
                dataId: this.dataId,
                data: this.data,
                change: this.change,
                isReadonly: this.isReadonly,
                x2oField: this.x2oField,
                x2oFieldId: this.x2oFieldId,
            },
            on: {
                changeView: this.changeView,
                updateDataIds: this.updateDataIds,
            },
        });
    },
    computed: {
        view () {
            const views = this.$store.state.data.views;
            if (this.viewId) return views[String(this.viewId)] || {};
            return {};
        },
        data () {
            if (this.model) {
                let data = this.$store.state.data.data;
                let changes = this.$store.state.data.changes;

                if (this.dataId) {
                    data = (data[this.model]) ? data[this.model][String(this.dataId)] || {} : {};
                    if (changes[this.model] && changes[this.model][String(this.dataId)] != undefined) changes = changes[this.model][String(this.dataId)];
                    else if (changes.new && changes.new[this.model] && changes.new[this.model][String(this.dataId)] != undefined) changes = changes.new[this.model][String(this.dataId)];
                    else changes = {};
                    return Object.assign({}, data, changes);
                }
                const d = {};
                _.each(this.dataIds, dataId => {
                    const _data = (data[this.model]) ? data[this.model][String(dataId)] || {} : {};
                    let _changes = {};
                    if (changes[this.model] && changes[this.model][String(dataId)] != undefined) _changes = changes[this.model][String(dataId)];
                    else if (changes.new && changes.new[this.model] && changes.new[this.model][String(dataId)] != undefined) _changes = changes.new[this.model][String(dataId)];
                    d[dataId] = Object.assign({}, _data, _changes);
                });
                return d;
            }
            return {};
        },
        change () {
            if (this.dataId) {
                const changes = this.$store.state.data.changes;
                if (changes[this.model] && changes[this.model][String(this.dataId)] != undefined) return changes[this.model][String(this.dataId)];
                else if (changes.new && changes.new[this.model] && changes.new[this.model][String(this.dataId)] != undefined) return changes.new[this.model][String(this.dataId)];
            }
            return {};
        },
    },
    methods: {
        changeView (viewId, dataId) {
            this.$emit('changeView', viewId, dataId);
        },
        updateDataIds (dataIds) {
            this.$emit('updateDataIds', dataIds);
        },
    },
});

export const FieldFormOne2Many = Vue.component('furet-ui-form-field-one2many', {
    mixins: [FormMixin, RelationShip],
    props: ['model', 'views', 'x2oField'],
    template: `
        <div v-if="this.isInvisible" />
        <b-tooltip 
            v-bind:label="getTooltip" 
            v-bind:position="tooltipPosition"
            v-bind:style="{'width': '100%'}"
            v-else
        >
            <b-field 
                v-bind:label="label"
                v-bind:type="getType"
                v-bind:message="getMessage"
                v-bind:style="{'width': 'inherit'}"
            >
                <furet-ui-x2m-view 
                    v-bind:model="model"
                    v-bind:views="views"
                    v-bind:viewId="viewId"
                    v-bind:dataIds="dataIds"
                    v-bind:dataId="dataId"
                    v-bind:isReadonly="isReadonly"
                    v-bind:x2oField="x2oField"
                    v-bind:x2oFieldId="config.dataId"
                    v-on:changeView="changeView"
                    v-on:updateDataIds="updateDataIds"
                />
            </b-field>
        </b-tooltip>`,
    data () {
        return {
            viewId: this.views && this.views[0] && this.views[0].viewId,
            dataId: null,
        }
    },
    created () {
        json_post('/field/x2m/get/views', {viewIds: _.map(this.views, v => v.viewId)}, {
            onSuccess: (result) => {
                dispatchAll(result);
            }
        });
    },
    computed: {
        dataIds () {
            return this.config && this.config.data && this.config.data[this.name] || [];
        },
        view () {
            return this.$store.state.data.view[this.viewId];
        },
    },
    methods: {
        changeView (viewId, dataId) {
            this.viewId = viewId;
            this.dataId = dataId;
        },
        updateDataIds (dataIds) {
            this.updateValue(dataIds)
        },
    },
})
