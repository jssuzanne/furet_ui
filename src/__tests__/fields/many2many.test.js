/**
This file is a part of the FuretUI project

   Copyright (C) 2017 Jean-Sebastien SUZANNE <jssuzanne@anybox.fr>

This Source Code Form is subject to the terms of the Mozilla Public License,
v. 2.0. If a copy of the MPL was not distributed with this file,You can
obtain one at http://mozilla.org/MPL/2.0/.
**/
import Vue from 'vue';
import Vuex from 'vuex';
import Buefy from 'buefy';
Vue.use(Buefy, {defaultIconPack: 'fa',});
import {store} from '../../store';
import {router} from '../../routes';
import {i18n} from '../../i18n';
import {FieldListMany2Many, FieldThumbnailMany2Many, FieldFormMany2Many} from '../../fields/relationship/many2many'

describe('Many2Many list component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
    });
    it('Render with data', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListMany2Many, {props: {
                row: {fieldname: ['1']},
                header: {name: 'fieldname', model: 'Test', display: 'fields.name'},
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test'}},
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render without data 1', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListMany2Many, {props: {
                row: {},
                header: {name: 'fieldname', model: 'Test', display: 'fields.name'},
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test'}},
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render without data 2', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListMany2Many, {props: {
                row: {fieldname: ['1']},
                header: {name: 'fieldname', model: 'Test', display: 'fields.name'},
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is True', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListMany2Many, {props: {
                row: {fieldname: ['1']},
                header: {name: 'fieldname', invisible: true, model: 'Test', display: 'fields.name'},
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test'}},
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListMany2Many, {props: {
                row: {fieldname: ['1']},
                header: {name: 'fieldname', invisible: false, model: 'Test', display: 'fields.name'},
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test'}},
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is true', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListMany2Many, {props: {
                row: {fieldname: ['1'], invisible: true},
                header: {name: 'fieldname', invisible: 'fields.invisible', model: 'Test', display: 'fields.name'},
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test'}},
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListMany2Many, {props: {
                row: {fieldname: ['1'], invisible: false},
                header: {name: 'fieldname', invisible: 'fields.invisible', model: 'Test', display: 'fields.name'},
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test'}},
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render fieldcolor', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldListMany2Many, {props: {
                row: {fieldname: ['1']},
                header: {name: 'fieldname', fieldcolor: 'color', model: 'Test', display: 'fields.name'},
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test', color: '#123456'}},
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
});

describe('Many2Many Thumbnail component', () => {
    const renderer = require('vue-server-renderer').createRenderer();
    beforeEach(() => {
        store.dispatch('UNITEST_CLEAR');
    });
    it('Render', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailMany2Many, {props: {
                data: {fieldname: ['1']},
                name: 'fieldname',
                model: 'Test',
                display: 'fields.name',
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test'}},
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render without data 1', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailMany2Many, {props: {
                data: {},
                name: 'fieldname',
                model: 'Test',
                display: 'fields.name',
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test'}},
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render without data 1', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailMany2Many, {props: {
                data: {fieldname: 1},
                name: 'fieldname',
                model: 'Test',
                display: 'fields.name',
            }}),
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is True', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailMany2Many, {props: {
                data: {fieldname: ['1']},
                name: 'fieldname',
                invisible: true,
                model: 'Test',
                display: 'fields.name',
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test'}},
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailMany2Many, {props: {
                data: {fieldname: ['1']},
                name: 'fieldname',
                invisible: false,
                model: 'Test',
                display: 'fields.name',
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test'}},
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is true', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailMany2Many, {props: {
                data: {fieldname: ['1'], invisible: true},
                name: 'fieldname',
                invisible: 'fields.invisible',
                model: 'Test',
                display: 'fields.name',
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test'}},
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render invisible condition is false', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailMany2Many, {props: {
                data: {fieldname: ['1'], invisible: false},
                name: 'fieldname',
                invisible: 'fields.invisible',
                model: 'Test',
                display: 'fields.name',
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test'}},
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render label', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailMany2Many, {props: {
                data: {fieldname: ['1']},
                name: 'fieldname',
                label: 'The label',
                model: 'Test',
                display: 'fields.name',
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test'}},
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render tooltip up', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailMany2Many, {props: {
                data: {fieldname: ['1']},
                name: 'fieldname',
                tooltip: 'The tooltip',
                model: 'Test',
                display: 'fields.name',
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test'}},
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render tooltip left', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailMany2Many, {props: {
                data: {fieldname: ['1']},
                name: 'fieldname',
                tooltip: 'The left tooltip',
                tooltip_position: 'is-left',
                model: 'Test',
                display: 'fields.name',
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test'}},
        });
        renderer.renderToString(vm, (err, str) => {
            expect(str).toMatchSnapshot();
        });
    });
    it('Render with fieldcolor', () => {
        const vm = new Vue({
            el: document.createElement('div'),
            store,
            router,
            i18n,
            render: h => h(FieldThumbnailMany2Many, {props: {
                data: {fieldname: ['1']},
                name: 'fieldname',
                model: 'Test',
                display: 'fields.name',
                fieldcolor: 'color',
            }}),
        });
        store.commit({
            type: 'UPDATE_DATA',
            model: 'Test',
            data: {'1': {name: 'Test', color: '#123456'}},
        });
        Vue.nextTick(() => {
            renderer.renderToString(vm, (err, str) => {
                expect(str).toMatchSnapshot();
            });
        });
    });
});

