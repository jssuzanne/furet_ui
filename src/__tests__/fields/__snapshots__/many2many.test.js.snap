// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Many2Many Thumbnail component Render 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!----> <span class="tag"><a>Test
                    </a></span>
<!---->
</div>
</span>
`;

exports[`Many2Many Thumbnail component Render invisible condition is false 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!----> <span class="tag"><a>Test
                    </a></span>
<!---->
</div>
</span>
`;

exports[`Many2Many Thumbnail component Render invisible condition is true 1`] = `<div data-server-rendered="true"></div>`;

exports[`Many2Many Thumbnail component Render invisible is True 1`] = `<div data-server-rendered="true"></div>`;

exports[`Many2Many Thumbnail component Render invisible is false 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!----> <span class="tag"><a>Test
                    </a></span>
<!---->
</div>
</span>
`;

exports[`Many2Many Thumbnail component Render label 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><label class="label">The label</label> <span class="tag"><a>Test
                    </a></span>
<!---->
</div>
</span>
`;

exports[`Many2Many Thumbnail component Render tooltip left 1`] = `
<span data-label="The left tooltip" data-server-rendered="true" class="tooltip is-primary is-left is-medium"><div class="field" style="width:inherit;"><!----> <span class="tag"><a>Test
                    </a></span>
<!---->
</div>
</span>
`;

exports[`Many2Many Thumbnail component Render tooltip up 1`] = `
<span data-label="The tooltip" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!----> <span class="tag"><a>Test
                    </a></span>
<!---->
</div>
</span>
`;

exports[`Many2Many Thumbnail component Render with fieldcolor 1`] = `
<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!----> <span class="tag" style="border:4px solid #123456;"><a>Test
                    </a></span>
<!---->
</div>
</span>
`;

exports[`Many2Many Thumbnail component Render without data 1 1`] = `<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!---->  <!----></div></span>`;

exports[`Many2Many Thumbnail component Render without data 1 2`] = `<span data-label="" data-server-rendered="true" class="tooltip is-primary is-top is-medium"><div class="field" style="width:inherit;"><!---->  <!----></div></span>`;

exports[`Many2Many list component Render fieldcolor 1`] = `
<div data-server-rendered="true"><span class="tag" style="border:4px solid #123456;"><a>Test
                </a></span></div>
`;

exports[`Many2Many list component Render invisible condition is false 1`] = `
<div data-server-rendered="true"><span class="tag"><a>Test
                </a></span></div>
`;

exports[`Many2Many list component Render invisible condition is true 1`] = `<span data-server-rendered="true"></span>`;

exports[`Many2Many list component Render invisible is True 1`] = `<span data-server-rendered="true"></span>`;

exports[`Many2Many list component Render invisible is false 1`] = `
<div data-server-rendered="true"><span class="tag"><a>Test
                </a></span></div>
`;

exports[`Many2Many list component Render with data 1`] = `
<div data-server-rendered="true"><span class="tag"><a>Test
                </a></span></div>
`;

exports[`Many2Many list component Render without data 1 1`] = `<div data-server-rendered="true"></div>`;

exports[`Many2Many list component Render without data 2 1`] = `<div data-server-rendered="true"></div>`;
